src/bapsicle.ts

```diff
@@ -0,0 +1,170 @@
+          case "STATUS":
+            // Bapsicle is telling us it's full state on this channel. Let's update the UI.
+
+            // TODO: This hangs the state of timers etc for a second or two, maybe do stuff more selectively
+            const channel = message.channel;
+            // Update the player state
+            const bapsicle_state = message.data;
+
+            const playerState: MixerState.PlayerStateEnum = bapsicle_state.paused
+              ? "paused"
+              : bapsicle_state.playing
+              ? "playing"
+              : "stopped";
+            store.dispatch(
+              MixerState.setPlayerState({ player: channel, state: playerState })
+            );
+
+            if (bapsicle_state.loaded_item) {
+              const raw_planitem = bapsicle_state.loaded_item;
+              var loadedItem: PlanItem = {
+                timeslotitemid: raw_planitem.timeslotItemId,
+                channel: message.channel,
+                type: raw_planitem.trackId ? "central" : "aux",
+                trackid: raw_planitem.trackId,
+                managedid: raw_planitem.managedId,
+                auxid: raw_planitem.managedId,
+                ...raw_planitem,
+              };
+              if (bapsicle_state.loaded) {
+                store.dispatch(MixerState.load(channel, loadedItem));
+              } else {
+                store.dispatch(
+                  MixerState.itemLoadPercentage({ player: channel, percent: 0 })
+                );
+              }
+            } else {
+              store.dispatch(MixerState.load(channel, null));
+            }
+
+            if (
+              !bapsicle_state.initialised ||
+              (bapsicle_state.loaded_item && !bapsicle_state.loaded)
+            ) {
+              store.dispatch(MixerState.itemLoadError({ player: channel }));
+            }
+
+            store.dispatch(
+              MixerState.setTimeLength({
+                player: channel,
+                time: bapsicle_state.length,
+              })
+            );
+            store.dispatch(
+              MixerState.setTimeCurrent({
+                player: channel,
+                time: bapsicle_state.pos_true,
+              })
+            );
+            store.dispatch(
+              MixerState.setAutoAdvance({
+                player: channel,
+                enabled: bapsicle_state.auto_advance,
+              })
+            );
+            store.dispatch(
+              MixerState.setPlayOnLoad({
+                player: channel,
+                enabled: bapsicle_state.play_on_load,
+              })
+            );
+            store.dispatch(
+              MixerState.setRepeat({
+                player: channel,
+                mode: bapsicle_state.repeat.toLowerCase(),
+              })
+            );
+
+            if (!("show_plan" in message.data)) {
+              console.error("Show plan data missing from status");
+              console.error(message.data);
```

Comment:
This will execute the next lines even if this is true. I don't know bapsicle's API well enough to know how likely this is, but that seems prone to runtime errors - should there be an `else` on this `if`?